src/socket/meson.build

```
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+# SPDX-FileCopyrightText: 2023-2024 Rachel Mant <git@dragonmux.network>
+# SPDX-FileContributor: Written by Rachel Mant <git@dragonmux.network>
+
+rSONSocketLibs = []
+if host_machine.system() == 'windows'
+	rSONSocketLibs += cxx.find_library('Ws2_32')
+endif
+
+rSONSocketSrc = ['socket.cxx']
+
+rSONSocket = library(
+	'rSON.socket',
+	rSONSocketSrc,
+	cpp_args: ['-DrSON_SOCKET_EXPORT_API'],
+	include_directories: [rSONIncludeDir, rSONSocketIncludeDir],
```

@dragonmux you can drop `rSONIncludeDir` by virtue of the dep:

```suggestion
	include_directories: [rSONSocketIncludeDir],
```

---

Good catch, thank you!